File: 3225-length-of-longest-subarray-with-at-most-k-frequency/length-of-longest-subarray-with-at-most-k-frequency.java
class Solution{
    public int maxSubarrayLength(int[] nums, int k){
        int n=nums.length;

        HashMap<Integer, Integer> mp=new HashMap<>();
        int res=0;
        int i=0, j=0;
        while(j<n){
            mp.put(nums[j], mp.getOrDefault(nums[j], 0)+1);

            while(mp.get(nums[j])>k && i<=j){
                mp.put(nums[i], mp.getOrDefault(nums[i], 0)-1);
                i++;
            }   

            res=Math.max(res, j-i+1);
            j++;
        }

        return res;


    }
}

File: 3225-length-of-longest-subarray-with-at-most-k-frequency/length-of-longest-subarray-with-at-most-k-frequency.py
class Solution:
    def maxSubarrayLength(self, nums: List[int], k: int) -> int:
        n=len(nums)

        mp=defaultdict(int)
        res=0
        i, j=0, 0
        while j<n:
            mp[nums[j]]+=1

            while mp[nums[j]]>k and i<=j:
                mp[nums[i]]-=1
                i+=1

            res=max(res, j-i+1)
            j+=1

        return res

